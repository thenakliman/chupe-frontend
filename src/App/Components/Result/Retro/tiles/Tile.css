@import "/src/App/styles/variables.css";

.tile {
  margin: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.tile .footer {
  display: flex;
  flex-direction: row;
  padding: 3px 4px;
  border-top: 1px solid black;
  font-size: 0.8rem;
}

.tile .footer__name {
  margin-left: 0.5rem;
}

.tile__description {
  padding: 0.7rem 4px;
  min-height: 5rem;
  min-width: 15rem;
  max-width: 30rem;
}

.footer__votes {
  margin-left: 0.61rem;
}

.footer__vote-text {
  cursor: pointer;
  font-weight: 600;
}