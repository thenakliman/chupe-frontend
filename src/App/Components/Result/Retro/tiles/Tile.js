import React from "react";

require('./Tile.css');

export const ActionItemTile = ({assignedTo, description}) => {
  return <div className={'tile u-brown-background-color u-white-font-color'}>
    <div className={'tile__description'}>
      {description}
    </div>
    <div className={'footer'}>
      <div>
        Assigned To:
      </div>
      <div className={'footer__name'}>
        {assignedTo}
      </div>
    </div>
  </div>
};

export const RetroPointTile = ({retroId, retroPointId, votes, description, className, onVoteCast}) => {
  return <div className={`tile u-white-font-color ${className}`}>
    <div className={'tile__description'}>
      {description}
    </div>
    <div className={'footer'}>
      <div className={'footer__vote-text'} onClick={() => onVoteCast(retroId, retroPointId)}>
        Vote
      </div>
      <div className={'footer__votes'}>
        {votes}
      </div>
    </div>
  </div>
};
