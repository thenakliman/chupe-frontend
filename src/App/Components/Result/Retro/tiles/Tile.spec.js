import React from "react";
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {ActionItemTile, RetroPointTile} from "./Tile";

describe('Show ActionItemTile', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ActionItemTile assignedTo={'assigned to'} description={'description'}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Show RetroPointTile', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<RetroPointTile
        retroId={12343}
        retroPointId={6543}
        votes={3}
        className={'u-red-background-color'}
        onVoteCast={jest.fn()}
        description={'some description'}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onVoteCast on click of vote', () => {
    const onVoteCast = jest.fn();
    const wrapper = shallow(<RetroPointTile
        retroId={12343}
        retroPointId={6543}
        votes={3}
        className={'u-red-background-color'}
        onVoteCast={onVoteCast}
        description={'some description'}/>);

    wrapper.find('.footer__vote-text').simulate('click');
    expect(onVoteCast).toHaveBeenCalledWith(12343, 6543);
  });
});
