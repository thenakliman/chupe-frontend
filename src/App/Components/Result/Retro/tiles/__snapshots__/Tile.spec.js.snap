// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show ActionItemTile should match snapshot 1`] = `
<div
  className="tile u-brown-background-color u-white-font-color"
>
  <div
    className="tile__description"
  >
    description
  </div>
  <div
    className="footer"
  >
    <div>
      Assigned To:
    </div>
    <div
      className="footer__name"
    >
      assigned to
    </div>
  </div>
</div>
`;

exports[`Show RetroPointTile should match snapshot 1`] = `
<div
  className="tile u-white-font-color u-red-background-color"
>
  <div
    className="tile__description"
  >
    some description
  </div>
  <div
    className="footer"
  >
    <div
      className="footer__vote-text"
      onClick={[Function]}
    >
      Vote
    </div>
    <div
      className="footer__votes"
    >
      3
    </div>
  </div>
</div>
`;
